shell/browser/api/electron_api_web_contents.cc

```diff
@@ -1383,13 +1383,19 @@ void WebContents::HandleNewRenderFrame(
       static_cast<content::RenderWidgetHostImpl*>(rwhv->GetRenderWidgetHost());
   if (rwh_impl)
     rwh_impl->disable_hidden_ = !background_throttling_;
-
-  WebFrameMain::RenderFrameCreated(render_frame_host);
 }
 
 void WebContents::RenderFrameCreated(
     content::RenderFrameHost* render_frame_host) {
   HandleNewRenderFrame(render_frame_host);
+
+  WebFrameMain::RenderFrameCreated(render_frame_host);
+
+  v8::Isolate* isolate = JavascriptEnvironment::GetIsolate();
+  v8::HandleScope handle_scope(isolate);
+  gin_helper::Dictionary details = gin_helper::Dictionary::CreateEmpty(isolate);
+  details.SetGetter("frame", render_frame_host);
```

Comment:
This seems dangerous. The `render_frame_host` pointer is not guaranteed to stay valid for the lifetime of this dictionary. Won't that potentially result in a use-after-free?

Reply:
That's a good point. I could probably substitute the pointer for a tuple of the process and frame ID as we emit in other WebContents events.

Reply:
Don't we already have a `WebFrameMain` object for this frame that we can attach directly, which already handles the lifetimes?

Reply:
All WebFrameMain instances are created on demand. The frame-created event is the first to emit a RFH directly which would elicit the creation of the corresponding WebFrameMain.

Reply:
See latest comment and 984c060.